src/common.rs

```diff
@@ -109,91 +109,47 @@ pub(crate) fn config_file_get() -> String {
-    match env::var("REVOCATION_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("general", "receive_revocation_port")
-            }
-        }
-        _ => config_get("general", "receive_revocation_port"),
-    }
+    config_get_env("general", "receive_revocation_port", "REVOCATION_PORT")
 }
 
 /// Returns cloud agent IP from keylime.conf if env var not present
 pub(crate) fn cloudagent_ip_get() -> Result<String> {
-    match env::var("CLOUDAGENT_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("cloud_agent", "cloudagent_ip")
-            }
-        }
-        _ => config_get("cloud_agent", "cloudagent_ip"),
-    }
+    config_get_env("cloud_agent", "cloudagent_ip", "CLOUDAGENT_IP")
 }
 
 /// Returns cloud agent port from keylime.conf if env var not present
 pub(crate) fn cloudagent_port_get() -> Result<String> {
-    match env::var("CLOUDAGENT_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("cloud_agent", "cloudagent_port")
-            }
-        }
-        _ => config_get("cloud_agent", "cloudagent_port"),
-    }
+    config_get_env("cloud_agent", "cloudagent_port", "CLOUDAGENT_PORT")
 }
 
 /// Returns registrar IP from keylime.conf if env var not present
 pub(crate) fn registrar_ip_get() -> Result<String> {
-    match env::var("REGISTRAR_IP") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("registrar", "registrar_ip")
-            }
-        }
-        _ => config_get("registrar", "registrar_ip"),
-    }
+    config_get_env("registrar", "registrar_ip", "REGISTRAR_IP")
 }
 
 /// Returns registrar port from keylime.conf if env var not present
 pub(crate) fn registrar_port_get() -> Result<String> {
-    match env::var("REGISTRAR_PORT") {
-        Ok(ip) => {
-            // The variable length must be larger than 0 to accept
-            if !ip.is_empty() {
-                Ok(ip)
-            } else {
-                config_get("registrar", "registrar_port")
-            }
-        }
-        _ => config_get("registrar", "registrar_port"),
+    config_get_env("registrar", "registrar_port", "REGISTRAR_PORT")
+}
+
+/// Returns the contact ip for the agent if set
+pub(crate) fn cloudagent_contact_ip_get() -> Option<String> {
+    match config_get_env("cloud_agent", "agent_contact_ip", "KEYLIME_AGENT_CONTACT_IP") {
+        Ok(ip) => Some(ip),
+        Err(_) => None // Ignore errors because this option might not be set
```

Comment:
Are you sure there could be a legitimate error here?

Reply:
The key might not be in `keylime.conf` and not specified as environment variable which is ok because this setting is optional. We currently have no distinction between error and key not found, so I didn't found a better way to handle that.

Reply:
ack, thanks!